### ISSUE TITLE
Air Shields spontaneously change air conditions

### ISSUE THREAD

**Issue by falcon2346** [d3athrow/vgstation13#35097]
#### Description of issue
When setting up an airshield in a room and then changing the room conditions on the air alarm the room will spontaneously reset itself, at least partway, to the original conditions. For instance, setting up an air shield and changing from human to vox will cause O2 to be siphoned out only to be spontaneously generated later, seemingly from nowhere. Removing the air shield fixes this issue, and if the airshield is installed after the atmosphere has already been changed it will no longer have this issue. 
Inorien can confirm this as well, as he witnessed it happening.

#### Difference between expected and actual behavior
Expected: The room become the desired conditions over time, regardless of if the door is open or not, as the shield blocks gas flow
Actual: The room will never reach the desired conditions, as the shield will cause the original gasses of the room to spontaneously generate.

#### Steps to reproduce
Put up airshield in a doorway, change settings to vox, marvel at cold fusion.

#### Length of time in which bug has been known to occur
Probably since airshields were added.

**Comment by Eneocho** [comment]
Airshields don't actually block gas flow, but rather freeze the atmospheric conditions inside it, such that it doesn't update when the rest of the area does, unless it is unpowered (meaning that the room might be breached but the one tile with the powered airshield does not lose moles, nor pressure). There's probably some mustard gas happening somewhere.

**Comment by pedr69** [comment]
![image](https://github.com/vgstation-coders/vgstation13/assets/18052467/dfad3deb-525d-4219-b77c-305fcf33caba)
scrubbers aren't very fast and the distro, which is what you're refilling the scrubbed oxygen with, still contains 20% oxygen
it'll take a while to get to pure nitrogen but it'll get there eventually

**Comment by Inorien** [comment]
Heres what I remember seeing Falcon do in the round:

![image](https://github.com/vgstation-coders/vgstation13/assets/16082007/179b5710-b832-4dd0-b73b-4bb18d3a0530)


correct me if this is wrong

**Comment by falcon2346** [comment]
> Heres what I remember seeing Falcon do in the round:
> 
> ![image](https://user-images.githubusercontent.com/16082007/271896188-179b5710-b832-4dd0-b73b-4bb18d3a0530.png)
> 
> correct me if this is wrong

This is correct, I was filtering out the oxygen from the line, and I even confirmed with a gas analyzer that the pipe leading to the room's vent contained 100% N2, yet O2 was still entering the room spontaneously. In case it matters, this was on the QM office in Waystation, but I experience something similar in patient room 1 of Box.

**Comment by Exxion** [comment]
Airshields are pretty straightforward. It's more likely to be caused by the airlock that's on the same tile.

EDIT: I can't seem to reproduce it.

**Comment by Inorien** [comment]
> Airshields are pretty straightforward. It's more likely to be caused by the airlock that's on the same tile.
> 
> EDIT: I can't seem to reproduce it.

I thought at the time that it might have been something to do with the airlock being there, I forget if the door was open/closed at any point during the airshield construction

I do remember looking at it with Falcon and being really perplexed though, I couldn't see how air would possibly be entering the room

**Comment by pedr69** [comment]
building the airshield inside a door that's closed sounds like it could potentially break stuff since you've got 2 stacked airtight objects
but when tested it functions normally
![image](https://github.com/vgstation-coders/vgstation13/assets/18052467/061703d5-bd24-4dfe-acb0-f31a28f08544)

**Comment by falcon2346** [comment]
Then I'm afraid I have no idea what caused it in these two specific cases. I'll see about recording the next time I adjust air conditions in a room.

**Comment by Inorien** [comment]
> Then I'm afraid I have no idea what caused it in these two specific cases. I'll see about recording the next time I adjust air conditions in a room.

@falcon2346 lets try getting a video recording next time, though i think we can leave this issue open - just because we can't trivially reproduce it doesn't mean that its a non-issue, we can expand here later

**Comment by falcon2346** [comment]
That's fair enough, I suppose there are enough issues that have been open for ages that having one more isn't going to hurt anything.
